### PR TITLE
remove unnecessary fields

### PR DESCRIPTION
**Pull request by deads2k**
Remove unnecessary fields and indirection from api server start

### CODE REVIEW

File: pkg/cmd/server/start/start_master.go [hunk 1]
```diff
@@ -540,32 +539,19 @@ func (m *Master) Start() error {
...
-		etcdClient, err := etcd.MakeEtcdClient(oc.Options.EtcdClientInfo)
-		if err != nil {
-			return err
-		}
-		if err := etcd.TestEtcdClient(etcdClient); err != nil {
-			return err
-		}
+	// TODO remove when this becomes a health check in 3.8
+	if err := testEtcdConnectivity(oc.Options.EtcdClientInfo); err != nil {
+		return nil
```
Comment (ironcladlou) [inline pkg/cmd/server/start/start_master.go]: `return err`?

Comment (deads2k) [inline pkg/cmd/server/start/start_master.go]: > return err?

uh awkward.

**Comment by ironcladlou** [comment]
Cool cleanup/simplification, /lgtm

**Comment by deads2k** [comment]
> Cool cleanup/simplification, /lgtm

@ironcladlou `/lgtm` has to be a on separate line

**Comment by stevekuznetsov** [comment]
Looks like retest was required

**Pull request merged**